docs/source/hsm.md

```diff
@@ -53,11 +53,12 @@ bccsp:
     Label: fabric
     hash: SHA2
     security: 256
+    Immutable: false
 ```
 
-You can also use environment variables to override the relevant fields of the
-configuration file. If you are connecting to an HSM using the Fabric CA server,
-you need to set the following environment variables:
+By default, when private keys are generated using the HSM, the private key is mutable, meaning PKCS11 private key  attributes can be changed after the key is generated. Setting `Immutable` to `true` means that the private key attributes cannot be altered after key generation. Before you configure immutability by setting `Immutable: true`, ensure that PKCS11 object cloning is supported by the HSM.
+
+You can also use environment variables to override the relevant fields of the configuration file. If you are connecting to softhsm2 using the Fabric CA server, you need to set the following environment variables:
```

Comment:
Because the first sentence implies the environment variables are specifically for overriding the config file, it might sound better to say something along the lines of `you may set the following environment variables`. Just something to better imply that the properties can both be set via the config and overwritten with the env vars. Otherwise reading this sentence might imply that you must configure it via the environment variables.

Reply:
Added both options throughout the content.